pkg/authorization/authz.go

```diff
@@ -54,13 +54,11 @@ type Ctx struct {
 // AuthZRequest authorized the request to the docker daemon using authZ plugins
 func (ctx *Ctx) AuthZRequest(w http.ResponseWriter, r *http.Request) error {
 	var body []byte
-	if sendBody(ctx.requestURI, r.Header) {
-		if r.ContentLength < maxBodySize {
-			var err error
-			body, r.Body, err = drainBody(r.Body)
-			if err != nil {
-				return err
-			}
+	if sendBody(ctx.requestURI, r.Header) && r.ContentLength > 0 && r.ContentLength < maxBodySize {
+		var err error
+		body, r.Body, err = drainBody(r.Body)
```

Comment:
Will this still truncate `r.Body` in the case where **size(r.Body) > maxBodySize**? Given that is handled by the case of `err == nil` in the reworked `drainBody` function, that return value is the new buffered reader/io.ReadCloser that is based on the `NewReaderSize` returned buffer of `maxBodySize`.  I added a comment in the test as well which may be clearer what I mean..